src/Footer.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { deleteTodos } from './api/todos';
+import { useTodo } from './TodoContext';
+import { ErrorStatus } from './types/Error';
+
+type Props = {
+  todos: Todo[],
+  active: number,
+  completed: number,
+  filter: FilterStatus,
+  setFilter(status: FilterStatus): void,
+  setTodos(todosArray: Todo[]): void,
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  active,
+  completed,
+  filter,
+  setFilter,
+  setTodos,
+}) => {
+  const { setError } = useTodo();
+  const removeCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setError(ErrorStatus.none);
+
+    try {
+      await Promise.all(completedTodos.map(todo => deleteTodos(todo.id)));
+
+      setTodos(
+        todos.filter(todo => !todo.completed),
+      );
+    } catch {
+      setError(ErrorStatus.none);
+    }
+  };
+
+  return (
+    <footer className="footer">
+      <span className="todo-count">
+        {active !== 1 ? `${active} items left` : `${active} item left`}
+      </span>
+
+      <ul className="filters">
+        <li>
+          <button
+            type="button"
+            onClick={(event) => {
+              event.preventDefault();
+              setFilter(FilterStatus.all);
+            }}
```

Comment:
Do not use anonymous function - create separate handler and pass reference here.
Fix all similar cases